harness/determined/common/schemas/expconf/_v0.py

```diff
@@ -221,58 +221,84 @@ def __init__(
 DataLayerConfigV0.finalize(DataLayerConfigV0_Type)
 
 
-class EnvironmentImageV0(schemas.SchemaBase):
+class CudaCompatMixin:
+    @no_type_check
+    def _ensure_cuda_compat(self) -> None:
+        # Backwards compatibility.
+        if self.cuda is None and self.gpu is not None:
+            self.cuda = self.gpu
+        # Equalize values.
+        self.gpu = self.cuda
+
+
+class EnvironmentImageV0(schemas.SchemaBase, CudaCompatMixin):
     _id = "http://determined.ai/schemas/expconf/v0/environment-image.json"
     cpu: Optional[str] = None
+    cuda: Optional[str] = None
     gpu: Optional[str] = None
+    rocm: Optional[str] = None
 
     @schemas.auto_init
     def __init__(
         self,
         cpu: Optional[str] = None,
+        cuda: Optional[str] = None,
         gpu: Optional[str] = None,
+        rocm: Optional[str] = None,
     ) -> None:
-        pass
+        self._ensure_cuda_compat()
 
     @classmethod
     def from_dict(cls, d: Union[dict, str], prevalidated: bool = False) -> "EnvironmentImageV0":
         # Accept either a string or a map of strings to strings.
         if isinstance(d, str):
-            d = {"cpu": d, "gpu": d}
-        return super().from_dict(d, prevalidated)
+            d = {"cpu": d, "cuda": d, "gpu": d, "rocm": d}
+        result = super().from_dict(d, prevalidated)
+        result._ensure_cuda_compat()
+        return result
 
     def runtime_defaults(self) -> None:
         if self.cpu is None:
             self.cpu = (
                 "determinedai/environments:py-3.8-pytorch-1.9-lightning-1.3-tf-2.4-cpu-825e8ee"
             )
-        if self.gpu is None:
-            self.gpu = (
-                "determinedai/environments:cuda-11.1-pytorch-1.9-lightning-1.3-tf-2.4-gpu-825e8ee"
-            )
+        if self.rocm is None:
+            self.rocm = "determinedai/environments:rocm-4.2-pytorch-1.9-tf-2.5-rocm-4537b95"
+
+        self.cuda = (
+            self.gpu
+            or "determinedai/environments:cuda-11.1-pytorch-1.9-lightning-1.3-tf-2.4-gpu-825e8ee"
+        )
+        self.gpu = self.cuda
```

Comment:
looks like you are still setting up the experiment config to be backwards compatible, but it's still dead code that still hasn't been given to users, so I don't think there's any need to.